warehouse/malware/utils.py

```diff
@@ -0,0 +1,42 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+
+from collections import defaultdict
+
+from warehouse.malware.models import (
+    MalwareCheck,
+    MalwareCheckObjectType,
+    MalwareCheckState,
+    MalwareCheckType,
+)
+
+
+@functools.lru_cache()
+def valid_check_types():
+    return set([t.value for t in MalwareCheckObjectType])
+
+
+def get_enabled_checks(session):
+    checks = (
+        session.query(MalwareCheck.name, MalwareCheck.hooked_object)
+        .filter(MalwareCheck.check_type == MalwareCheckType.event_hook)
+        .filter(MalwareCheck.state == MalwareCheckState.enabled)
+        .all()
+    )
+    results = defaultdict(list)
```

Comment:
Aha, I think this negates my concern about a KeyError in the event listener :)

Reply:
Yeah, the `defaultdict` will just return an empty list for a missing key, so we should raise a `KeyError` on `enabled_checks[obj.__class__.__name__]`.

Reply:
Using a defaultdict here certainly makes what's going on in the event listener a bit implicit, but it works. I'm torn on doing it this way versus filtering in `determine_malware_checks`.

Reply:
I'll update it to use `.get` instead.